site24x7/websitemonitor.go

```diff
@@ -207,199 +238,72 @@ func websiteMonitorCreateOrUpdate(method, url string, expectedResponseStatus int
 	}
 
 	i := 0
-	actionRefs := make([]ActionRef, len(d.Get("actions").(map[string]interface{})))
-	for k, v := range d.Get("actions").(map[string]interface{}) {
+	actionRefs := make([]api.ActionRef, len(d.Get("action_ids").(map[string]interface{})))
+	for k, v := range d.Get("action_ids").(map[string]interface{}) {
 		tmp, err := strconv.Atoi(k)
 		if err != nil {
 			panic(err)
 		}
-		alertType := Status(tmp)
-		actionRefs[i] = ActionRef{ActionID: v.(string), AlertType: alertType}
+		alertType := api.Status(tmp)
+		actionRefs[i] = api.ActionRef{ActionID: v.(string), AlertType: alertType}
 		i++
 	}
 
-	m := &WebsiteMonitor{
+	return &api.Monitor{
+		MonitorID:             d.Id(),
 		DisplayName:           d.Get("display_name").(string),
 		Type:                  "URL",
 		Website:               d.Get("website").(string),
 		CheckFrequency:        strconv.Itoa(d.Get("check_frequency").(int)),
 		HTTPMethod:            d.Get("http_method").(string),
 		AuthUser:              d.Get("auth_user").(string),
 		AuthPass:              d.Get("auth_pass").(string),
-		MatchingKeyword:       new(ValueAndSeverity),
-		UnmatchingKeyword:     new(ValueAndSeverity),
-		MatchRegex:            new(ValueAndSeverity),
+		MatchingKeyword:       new(api.ValueAndSeverity),
+		UnmatchingKeyword:     new(api.ValueAndSeverity),
+		MatchRegex:            new(api.ValueAndSeverity),
 		MatchCase:             d.Get("match_case").(bool),
 		UserAgent:             d.Get("user_agent").(string),
 		CustomHeaders:         customHeaders,
 		Timeout:               d.Get("timeout").(int),
 		LocationProfileID:     d.Get("location_profile_id").(string),
-		NotificationProfileID: d.Get("notification_profile_id").(string),
-		ThresholdProfileID:    d.Get("threshold_profile_id").(string),
+		NotificationProfileID: d.Get("NotificationProfileID").(string),
+		ThresholdProfileID:    d.Get("ThresholdProfileID").(string),
```

Comment:
I think you want `"threshold_profile_id"` here.

Reply:
Done